$MirOS$

	The m4_HAS_AC_CT_FOURARGS hunk solves a local problem (that we have
	a libtool which works with autoconf 2.13 as well as 2.59+ and thus
	define m4_version_compare even for the former).

	The other hunks are suggested for upstream inclusion and fix issues
	with SSL detection.

--- aclocal.m4.orig	Mon Aug 13 14:13:30 2007
+++ aclocal.m4	Mon Aug 13 14:18:42 2007
@@ -1408,7 +1408,7 @@ AC_DEFUN([CF_CHECK_SSL_X509],[
 AC_MSG_CHECKING(for X509 support)
 AC_TRY_LINK(CF__SSL_HEAD [
 #include <openssl/x509.h>],
-	[X509_verify_cert_error_string(X509_STORE_CTX_get_error(X509_STORE_CTX *0))],
+	[X509_verify_cert_error_string(X509_STORE_CTX_get_error(NULL))],
 	[cf_x509_support=yes],
 	[cf_x509_support=no])
 AC_MSG_RESULT($cf_x509_support)
@@ -1425,9 +1425,12 @@ dnl deliberate incompatibility.
 dnl	$1 = name of type to check for
 dnl	$2 = default type
 dnl	$3 = additional #include's and related preprocessor lines.
+ifdef([m4_HAS_AC_CT_FOURARGS], [m4_undefine([m4_HAS_AC_CT_FOURARGS])])dnl
+ifelse(m4_PACKAGE_VERSION, [fnord_acsalt], [],
+[ifdef([m4_version_compare],[m4_define([m4_HAS_AC_CT_FOURARGS])])])dnl
 AC_DEFUN([CF_CHECK_TYPE],
 [
-ifdef([m4_version_compare],[
+ifdef([m4_HAS_AC_CT_FOURARGS],[
 	AC_CHECK_TYPE([$1],ac_cv_type_$1=yes,ac_cv_type_$1=no,[$3])
 	],[
 	AC_MSG_CHECKING(for $1)
@@ -4247,11 +4250,11 @@ AC_DEFUN([CF_SSL],[
       esac
 
     fi
-    CF_CHECK_SSL_X509
     if test -n "$cf_cv_library_path_ssl" ; then
       CF_ADD_LIBDIR($cf_cv_library_path_ssl)
     fi
     LIBS="-lssl -lcrypto $LIBS"
+    CF_CHECK_SSL_X509
   fi
 ])dnl
 dnl ---------------------------------------------------------------------------
